src/iadmin.cpp

```diff
@@ -143,6 +149,68 @@ auto get_replica_value(
     return v;
 } // get_replica_value
 
+auto print_delay_server_info() -> int
+{
+    try {
+        namespace adm = irods::experimental::administration;
+
+        const auto user_type = adm::client::type(*Conn, adm::user{Conn->clientUser.userName});
+        
+        if (!user_type) {
+            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
+        }
+
+        if (*user_type != adm::user_type::rodsadmin) {
+            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
+        }
+    }
+    catch (const irods::exception& e) {
+        std::cerr << "ERROR: " << e.client_display_what() << '\n'; // TODO Is there a common format for printing error messages?
+        return e.code();
+    }
+    catch (const std::exception& e) {
+        std::cerr << "ERROR: " << e.what() << '\n';
+        return 1; // TODO What error code should we use here?
```

Comment:
seems okay

Reply:
Yeah, I guess this is correct. The OS doesn't know what the iRODS error code is. We can print the iRODS error code to the terminal and return a non-zero value for these cases.